source/core/Array.cpp

```diff
@@ -1211,8 +1211,11 @@ VIREO_FUNCTION_SIGNATURE7(ArrayDeleteND, TypedArrayCoreRef, StaticType, void,
 
     IntIndex startIndex = offset > 0 ? offset : 0;
     IntIndex endIndex = offset +
-        deletedPortionLength > arrayIn->GetLength(numberOfDimensions-1-dimensionToDelete) ?
+        deletedPortionLength > arrayIn->GetLength(numberOfDimensions-1-dimensionToDelete) - startIndex ?
         arrayIn->GetLength(numberOfDimensions-1-dimensionToDelete) : offset + deletedPortionLength;
```

Comment:
It appears the code is adding offset twice in the false clause of the conditional operator (?). I think that section is incorrect and it should just use "deletedPortionLength" and not "offset + deletedPortionLenght". I wonder if we have tests that exercise this scenario.

Reply:
Also shouldn't "arrayIn->GetLength(numberOfDimensions-1-dimensionToDelete)" in the true clause of the conditional operator(?) also use - startIndex? Otherwise endIndex would be ending beyond the desired limits.

Reply:
The code could probably be changed to make this more clear, but it is looking at whether offset + deletedPortionLength is greater than array length - startIndex. It doesn't add offset to whatever is returned from the conditional. endIndex gets assigned to what is returned. I could add parentheses to make it more clear, though.

Also, we want endIndex to equal the length of the array in the true case, which is why we don't subtract startIndex.

Reply:
Ok, we should make the code to be clearer. We can do this later.